pkg/apis/extensions/v1beta1/types.go

```diff
@@ -217,6 +217,24 @@ type DeploymentSpec struct {
 	// Indicates that the deployment is paused and will not be processed by the
 	// deployment controller.
 	Paused bool `json:"paused,omitempty"`
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	RollbackTo *RollbackConfig `json:"rollbackTo,omitempty"`
+}
+
+// DeploymentRollback stores the information required to rollback a deployment.
+type DeploymentRollback struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Required: This must match the Name of a deployment.
+	Name string `json:"name"`
+	// The annotations to be updated to a deployment
+	UpdatedAnnotations map[string]string `json:"updatedAnnotations,omitempty"`
+	// The config of this deployment rollback.
+	RollbackTo RollbackConfig `json:"rollbackTo"`
+}
+
+type RollbackConfig struct {
+	// The revision to rollback to. If set to 0, rollbck to the last revision.
+	Revision int32 `json:"revision,omitempty"`
```

Comment:
The general reason, of course, is so that there's no chance of overflow.

In this particular case, that may seem unlikely, however (a) internally we have applications that run for years that are updated by automated processes, which sometimes do unexpected things, and (b) it's easier to make a blanket convention than to evaluate such things on a case-by-case basis.